### ISSUE TITLE
ModuleNotFoundError: No module named 'sparse_dot_topn.sparse_dot_topn

### ISSUE THREAD

**Issue by AlfaroDanielV** [ing-bank/sparse_dot_topn#124]
### Describe your issue.

I get this error whrn trying to import sparse_dot_topn.sparse_dot_topn as spt. I already have read the readme. Im on a Mac M3, 

### Reproducing Code Example

```python
##I have try to install it using this 2 approaches: 
pip install sparse_dot_topn
pip install sparse_dot_topn --no-binary sparse_dot_topn 

##Both run without issue but then get prompt the module error when trying to use the import
```

### Error message

```shell
ModuleNotFoundError: No module named 'sparse_dot_topn.sparse_dot_topn
```

### SciPy/NumPy/Python version and system information

```shell
1.14.1 2.1.3 1.1.5 sys.version_info(major=3, minor=12, micro=4, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: extraframeworks
    found: true
    include directory: unknown
    lib directory: unknown
    name: accelerate
    openblas configuration: unknown
    pc file directory: unknown
    version: unknown
  lapack:
    detection method: extraframeworks
    found: true
    include directory: unknown
    lib directory: unknown
    name: accelerate
    openblas configuration: unknown
    pc file directory: unknown
    version: unknown
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.12.0
Compilers:
  c:
    commands: cc
    linker: ld64
    name: clang
    version: 15.0.0
  c++:
    commands: c++
    linker: ld64
    name: clang
    version: 15.0.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.11
  fortran:
    commands: gfortran
    linker: ld64
    name: gcc
    version: 13.3.0
  pythran:
    include directory: ../../../../../../private/var/folders/hw/1f0gcr8d6kn9ms0_wn0_57qc0000gn/T/pip-build-env-bjescfad/overlay/lib/python3.12/site-packages/pythran
    version: 0.16.1
Machine Information:
  build:
    cpu: aarch64
    endian: little
    family: aarch64
    system: darwin
  cross-compiled: false
  host:
    cpu: aarch64
    endian: little
    family: aarch64
    system: darwin
Python Information:
  version: '3.12'
```

**Comment by RUrlus** [comment]
Hi @AlfaroDanielV,

```python
import sparse_dot_topn.sparse_dot_topn as spt
```
is not expected to work, as there is no module `sparse_dot_topn` within the package.
We use a top-level structure where everything should be imported from the package, like

```python
from sparse_dot_topn import sp_matmul_topn
```
or you could use

```python
import sparse_dot_topn as spt
...
spt.sp_matmul_topn(...)
```

Why do you expect this structure? I checked but don't see any code in the readme that suggests this should work.

**Comment by AlfaroDanielV** [comment]
Thank you!  the confusion might stem from an older issue with the same problem on this repo, which was resolved by following the installation instructions in the README. That’s why I specified that I’ve already gone through the README.

I was tasked with refactoring a pipeline that already had a notebook using this code, and it was crashing due to the import error. I will continue using sp_matmul_topn as suggested.

Best regards! 🇨🇷
